Source/Diagnostics/ReducedDiags/RawBFieldReduction.H

```diff
@@ -0,0 +1,281 @@
+            const amrex::MultiFab &By = warpx.getBfield(lev,1);
+            const amrex::MultiFab &Bz = warpx.getBfield(lev,2);
+
+            constexpr int index_Bx = 0;
+            constexpr int index_By = 1;
+            constexpr int index_Bz = 2;
+            amrex::IntVect Bx_nodalType = Bx.ixType().toIntVect();
+            amrex::IntVect By_nodalType = By.ixType().toIntVect();
+            amrex::IntVect Bz_nodalType = Bz.ixType().toIntVect();
+
+            amrex::ReduceOps<ReduceOp> reduceBx_op;
+            amrex::ReduceOps<ReduceOp> reduceBy_op;
+            amrex::ReduceOps<ReduceOp> reduceBz_op;
+            amrex::ReduceData<amrex::Real> reduceBx_data(reduceBx_op);
+            amrex::ReduceData<amrex::Real> reduceBy_data(reduceBy_op);
+            amrex::ReduceData<amrex::Real> reduceBz_data(reduceBz_op);
+
+            using ReduceTuple = typename decltype(reduceBx_data)::Type;
+
+            amrex::Geometry const & geom = warpx.Geom(lev);
+            const amrex::RealBox& real_box = geom.ProbDomain();
+            const auto dx = geom.CellSizeArray();
+
+#ifdef AMREX_USE_OMP
+#pragma omp parallel if (amrex::Gpu::notInLaunchRegion())
+#endif
+            for ( amrex::MFIter mfi(Bx, false); mfi.isValid(); ++mfi)
+            {
+                const amrex::Box& tx = mfi.tilebox(Bx_nodalType);
+                const amrex::Box& ty = mfi.tilebox(By_nodalType);
+                const amrex::Box& tz = mfi.tilebox(Bz_nodalType);
+
+                const amrex::IntVect lx = tx.smallEnd();
+                const amrex::IntVect hx = tx.bigEnd();
+
+                const amrex::IntVect ly = ty.smallEnd();
+                const amrex::IntVect hy = ty.bigEnd();
+
+                const amrex::IntVect lz = tz.smallEnd();
+                const amrex::IntVect hz = tz.bigEnd();
+
+                const auto& Bx_arr = Bx[mfi].array();
+                const auto& By_arr = By[mfi].array();
+                const auto& Bz_arr = Bz[mfi].array();
+
+                reduceBx_op.eval(tx, reduceBx_data,
+                [=] AMREX_GPU_DEVICE (int i, int j, int k) ->ReduceTuple
+                {
+                    // Shift x, y, z position based on index type
+                    amrex::Real fac_x = (1._rt - Bx_nodalType[0]) * dx[0] * 0.5_rt;
+                    amrex::Real x = i * dx[0] + real_box.lo(0) + fac_x;
+#if (AMREX_SPACEDIM==2)
+                    amrex::Real y = 0._rt;
+                    amrex::Real fac_z = (1._rt - Bx_nodalType[1]) * dx[1] * 0.5_rt;
+                    amrex::Real z = j * dx[1] + real_box.lo(1) + fac_z;
+#else
+                    amrex::Real fac_y = (1._rt - Bx_nodalType[1]) * dx[1] * 0.5_rt;
+                    amrex::Real y = j * dx[1] + real_box.lo(1) + fac_y;
+                    amrex::Real fac_z = (1._rt - Bx_nodalType[2]) * dx[2] * 0.5_rt;
+                    amrex::Real z = k * dx[2] + real_box.lo(2) + fac_z;
+#endif
+                    // we want to weight interior face-nodes by 1
```

Comment:
I would replace all instances of "face-nodes" with simply "faces".  Including the word 'node' in there could be misleading.  Something that is nodal in exactly one direction is "face" data; but in AMReX-speak we just refer to these as "faces".